src/main/java/org/openhab/tools/analysis/checkstyle/api/AbstractStaticCheck.java

```diff
@@ -40,38 +43,65 @@
      * returns the line number, where the text was found
      *
-     * @param fileContent - each element of the list represents a line from the file
+     * @param fileText - represents the content of a file
      * @param searchedText - the text that we are looking for
      * @param startLineNumber - the line number from which the search starts exclusive, to start the
      *            search of the beginning of the text the startLineNumber should be 0
      * @return the number of the line starting from 1, where the searched text occurred for the first
-     *         time, or -1 if
-     *         no match was found
+     *         time
+     * @throws NoResultException when no match was found
      */
-    protected int findLineNumber(String[] fileContent, String searchedText, int startLineNumber) {
-        for (int lineNumber = startLineNumber; lineNumber < fileContent.length; lineNumber++) {
-            String line = fileContent[lineNumber];
+    protected int findLineNumber(FileText fileText, String searchedText, int startLineNumber) throws NoResultException {
+        for (int lineNumber = startLineNumber; lineNumber < fileText.size(); lineNumber++) {
+            String line = fileText.get(lineNumber);
             if (line.contains(searchedText)) {
                 // The +1 is to compensate the 0-based list and the 1-based text file
                 return lineNumber + 1;
             }
         }
-        return -1;
+        String message = MessageFormat.format(
+                "`{0}` was not found in the file {1} starting from line `{2}`."
+                        + " Check if it is split between multiple lines or it is missing",
+                searchedText, fileText.getFile().getAbsolutePath(), startLineNumber);
+        throw new NoResultException(message);
+    }
+
+    /**
+     * Finds the first occurrence of a text in a list of text lines representing the file content and
+     * returns the line number, where the text was found
+     *
+     * @param fileText - represents the content of a file
+     * @param searchedText - the text that we are looking for
+     * @param startLineNumber - the line number from which the search starts exclusive, to start the
+     *            search of the beginning of the text the startLineNumber should be 0
+     * @param warningMessage - message to be logged as warning in case no match is found
+     * @return the number of the line starting from 1, where the searched text occurred for the first
+     *         time, or 0 if no matches are found
+     */
+    protected int findLineNumber(FileText fileText, String searchedText, int startLineNumber, String warningMessage) {
+        try {
+            return findLineNumber(fileText, searchedText, startLineNumber);
+        } catch (NoResultException e) {
+            logger.warn(warningMessage + " Fall back to 0.", e);
+            return 0;
```

Comment:
Why do you overload this method with one that behaves differently, if you are really that you want this giving them separate name would imho be more appropriate.

Reply:
Good point, I have changed the name